yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -125,20 +125,9 @@ type ParserIssueDetector struct {
 	//column is the key (qualifiedTableName.column_name) -> column stats
 	columnStatistics map[string]utils.ColumnStatistics
 
-	// Pre-computed index and prefix coverage map for missing foreign key index detection.
-	// Stores all possible column combinations that are covered by indexes, including prefixes.
-	//
-	// Examples:
-	// - Index on (user_id, product_id, order_date) creates entries:
-	//   "orders.user_id" = true
-	//   "orders.user_id,product_id" = true
-	//   "orders.user_id,product_id,order_date" = true
-	// - Foreign key on (user_id, product_id) is detected as covered
-	// - Primary key on (id) creates entry: "users.id" = true
-	// - Unique constraint on (email, status) creates entries:
-	//   "users.email" = true
-	//   "users.email,status" = true
-	indexPrefixCoverageMap map[string]bool
+	// Indexes stored per table
+	// Key is table name, value is slice of indexes on that table.
+	tableIndexes map[string][]*queryparser.Index
```

Comment:
Just to elaborate, I was having a discussion with Priyanshi yesterday - 
It will be useful to store it like this (i.e. generic list of indexes) so that in future, if we want to extract any other information out of these indexes, we can re-use the same field. Alternatively, if we stored a very specific set of values here that is only useful for our current task (of checking fk index coverage), that wouldn't be very generically useful in the future.